File: backend/Descansario.Api/Models/Vacation.cs
namespace Descansario.Api.Models;

public class Vacation
{
    public int Id { get; set; }

    public int PersonId { get; set; }

    public DateTime StartDate { get; set; }

    public DateTime EndDate { get; set; }

    // WorkingDaysCount ya NO se persiste - se calcula on-demand con WorkingDaysCalculator
    // Esto asegura que siempre esté actualizado si cambian feriados

    public VacationStatus Status { get; set; } = VacationStatus.Pending;

    public string? Notes { get; set; } // Soporte para markdown

    // Navigation property
    public Person? Person { get; set; }
}

public enum VacationStatus
{
    Pending,
    Approved,
    Rejected
}

File: frontend/src/lib/types/vacation.ts
export type VacationStatus = 'Pending' | 'Approved' | 'Rejected';

export interface Vacation {
  id: number;
  personId: number;
  personName: string;
  startDate: string; // ISO date string
  endDate: string; // ISO date string
  workingDaysCount: number;
  status: VacationStatus;
}

export interface VacationFormData {
  personId: number;
  startDate: string; // ISO date string
  endDate: string; // ISO date string
  status?: VacationStatus;
}

export type CreateVacationDto = VacationFormData;
export type UpdateVacationDto = VacationFormData;
